### PR TITLE
[NEW] Add multiple language support - text messages

### PR DESCRIPTION
**Pull request by murtaza98**
**This feature in action:** (Here I've selected Hindi Language)

Screenshot from 2020-10-04 23-34-25

### CODE REVIEW

File: config/Settings.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export enum AppSetting {
     DialogflowServiceUnavailableMessage = 'dialogflow_service_unavailable_message',
     DialogflowCloseChatMessage = 'dialogflow_close_chat_message',
     DialogflowHideQuickReplies = 'dialogflow_hide_quick_replies',
+    DialogflowLanguage = 'dialogflow_language',
```
Comment (renatobecker) [inline config/Settings.ts:14]:
```suggestion
    DialogflowDefaultLanguage = 'dialogflow_default_language',
```

File: config/Settings.ts [hunk 2]
```diff
@@ -19,6 +20,36 @@ export enum DefaultMessage {
     DEFAULT_DialogflowCloseChatMessage = 'Closing the chat, Goodbye',
 }
 
+export const LanguageCode = [
+    { key: 'zh-CN', i18nLabel: 'chinese_simplified' },
```
Comment (renatobecker) [inline config/Settings.ts:24]:
```suggestion
    { key: 'zh-CN', i18nLabel: 'Chinese Simplified' },
```

File: config/Settings.ts [hunk 3]
```diff
@@ -19,6 +20,36 @@ export enum DefaultMessage {
...
+    { key: 'da', i18nLabel: 'danish' },
+    { key: 'nl', i18nLabel: 'dutch' },
+    { key: 'en', i18nLabel: 'english' },
+    { key: 'en-AU', i18nLabel: 'english_australia' },
```
Comment (renatobecker) [inline config/Settings.ts:28]:
```suggestion
    { key: 'en-AU', i18nLabel: 'Australian English' },
```

File: config/Settings.ts [hunk 4]
```diff
@@ -19,6 +20,36 @@ export enum DefaultMessage {
     DEFAULT_DialogflowCloseChatMessage = 'Closing the chat, Goodbye',
 }
 
+export const LanguageCode = [
+    { key: 'zh-CN', i18nLabel: 'chinese_simplified' },
+    { key: 'da', i18nLabel: 'danish' },
```
Comment (renatobecker) [inline config/Settings.ts:25]: Let's do the same for the rest of the languages below.

File: config/Settings.ts [hunk 5]
```diff
@@ -71,6 +102,16 @@ export const settings: Array<ISetting> = [
         i18nDescription: 'target_department_for_handover_description',
         required: false,
     },
+    {
+        id: AppSetting.DialogflowLanguage,
```
Comment (renatobecker) [inline config/Settings.ts:106]:
```suggestion
        id: AppSetting.DialogflowDefaultLanguage,
```

File: lib/Dialogflow.ts [hunk 6]
```diff
@@ -20,9 +20,11 @@ class DialogflowClass {
                              requestType: DialogflowRequestType): Promise<IDialogflowMessage> {
         const serverURL = await this.getServerURL(read, modify, http, sessionId);
 
+        const languageCode = await getAppSettingValue(read, AppSetting.DialogflowLanguage);
```
Comment (renatobecker) [inline lib/Dialogflow.ts:23]:
```suggestion
        const languageCode = await getAppSettingValue(read, AppSetting.DialogflowDefaultLanguage);
```

File: config/Settings.ts [hunk 7]
```diff
@@ -71,6 +102,16 @@ export const settings: Array<ISetting> = [
...
+        public: true,
+        type: SettingType.SELECT,
+        values: LanguageCode,
+        packageValue: 'en',
+        value: 'en',
+        i18nLabel: 'dialogflow_language',
```
Comment (renatobecker) [inline config/Settings.ts:112]:
```suggestion
        i18nLabel: 'dialogflow_default_language',
```

**Review by renatobecker — DISMISSED** [review]

**Review by cauefcr — APPROVED** [review]

**Pull request merged**